src/main/java/gift/controller/AdminController.java

```diff
@@ -19,44 +23,60 @@
 @RequestMapping("/admin/list")
 public class AdminController {
 
-    private ProductDao productDao;
+    private final ProductDao productDao;
+    private final ProductValidator productValidator;
 
-    @Autowired
-    public void setProductDao(ProductDao productDao) {
+    public AdminController(ProductDao productDao, ProductValidator productValidator) {
         this.productDao = productDao;
+        this.productValidator = productValidator;
     }
 
     @GetMapping
     public String getAllProducts(Model model) {
         List<ProductResponseDto> productList = productDao.selectAllProduct()
             .stream()
-            .map(Product::toProductResponseDto)
+            .map(ProductResponseDto::from)
             .collect(Collectors.toList());
```

Comment:
줄여서 한 번에 사용도 가능합니다 :) 
```suggestion
            .toList();
```